tests/test_config.py

```diff
@@ -10,128 +20,41 @@
-            },
-            "SE-SE1": {},
-            "SE-SE2": {},
-            "SE-SE3": {},
-            "SE-SE4": {},
-        }
-        zone_neighbours = config.generate_zone_neighbours(zones, exchanges)
-        self.assertDictEqual(
-            zone_neighbours,
-            {
-                "NO-NO1": ["SE-SE3"],
-                "NO-NO3": ["SE-SE2"],
-                "NO-NO4": ["SE-SE1", "SE-SE2"],
-                "SE-SE1": ["NO-NO4"],
-                "SE-SE2": ["NO-NO3", "NO-NO4"],
-                "SE-SE3": ["NO-NO1"],
-            },
-        )
-
-    def test_generate_zone_neighbours_two_subzones_from_same(self):
-        exchanges = {
-            "SE-SE1->SE-SE2": {"parsers": {"exchange": "source"}},
-        }
-        zones = {
-            "SE": {
-                "subZoneNames": ["SE-SE1", "SE-SE2", "SE-SE3", "SE-SE4"],
-            },
-            "SE-SE1": {},
-            "SE-SE2": {},
-            "SE-SE3": {},
-            "SE-SE4": {},
-        }
-        zone_neighbours = config.generate_zone_neighbours(zones, exchanges)
-        self.assertDictEqual(
-            zone_neighbours,
-            {"SE-SE1": ["SE-SE2"], "SE-SE2": ["SE-SE1"]},
-        )
-
-    def test_generate_zone_neighbours_GB(self):
-        # That's an interesting case as GB has islands, which are not subzones
-        # It means that GB->GB-NIR are valid exchanges and that
-        # GB and GB-NIR are neighbours
-        exchanges = {
-            "GB->GB-NIR": {"parsers": {"exchange": "source"}},
-            "GB->GB-ORK": {"parsers": {"exchange": "source"}},
-        }
-        zones = {
-            "GB": {},
-            "GB-NIR": {},
-            "GB-ORK": {},
-        }
-        zone_neighbours = config.generate_zone_neighbours(zones, exchanges)
-        self.assertDictEqual(
-            zone_neighbours,
-            {"GB": ["GB-NIR", "GB-ORK"], "GB-NIR": ["GB"], "GB-ORK": ["GB"]},
-        )
-
-    def test_generate_zone_neighbours_no_exchange_parser(self):
-        exchanges = {
-            "DE->FR": {"parsers": {}},
-        }
-        zones = {
-            "DE": {},
-            "FR": {},
-        }
-        zone_neighbours = config.generate_zone_neighbours(zones, exchanges)
-        self.assertDictEqual(zone_neighbours, {})
-
-    def test_ZONE_NEIGHBOURS(self):
-        zone_neighbours = config.generate_zone_neighbours(
-            config.ZONES_CONFIG, config.EXCHANGES_CONFIG
-        )
-        self.assertIn("DK-DK1", zone_neighbours.keys())
-        dk_neighbours = zone_neighbours["DK-DK1"]
-
+    def test_zones_config(self):
+        # ZONES_CONFIG is a dict mapping zones to their config,
+        # which includes lots of keys with information about the zone.
+        self.assertIn("DK-DK1", config.ZONES_CONFIG.keys())
+        self.assertIn("bounding_box", config.ZONES_CONFIG["DK-DK1"].keys())
+        self.assertIn("capacity", config.ZONES_CONFIG["DK-DK1"].keys())
```

Comment:
We can not use string literals here when the expected type is ZoneKey. The same goes for the functions further down as well.

Reply:
I have a similar question as above -- should I use typing.cast or something else?

Is this actually checked anywhere? AFAIU ZoneKey is an alias for string, so what is meant exactly by "we can not use string literals here"? (The tests do pass with no warnings.)

Reply:
The ZoneKey is an extension of the string type, which means all ZoneKeys are strings but not all strings are ZoneKeys basically.

Right now it's basically the same thing though but type checking tools will fail on it.